### PR TITLE
Adds Folder placement for generated code and File Per Entity generation.

### PR DESCRIPTION
**Pull request by jtolar**
## Breaking change

No Breaking Changes.

## Proposed change

- Add's the ability to specify a folder for the generated code to be placed in using a command line switch (-f) to specify a folder name. 
- Add's the ability to generate separate files for each Interface/Class/Record generated.

Added -fpe command line switch (no value to set) to activate the File Per Entity option. If -fpe switch is specified the -o argument is ignored and will create a separate file using the name of the interface/class/record.

Added -f to specify a directory name to be created and which to place the generated file or files. 

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ X] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: No
- This PR is related to issue: No
- Link to documentation pull request: 

## Checklist

- [x ] The code change is tested and works locally.
- [x ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ x] There is no commented out code in this PR.
- [ x] I have followed the [development checklist][dev-checklist]
- [ x] The code compiles without warnings (code quality chek)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [x] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/net-daemon/docs

### CODE REVIEW

**Comment by FrankBakkerNl** [comment]
Cool feature, I dont have time to do a good review until next week. Maybe @helto4real can check it out?

**Comment by jtolar** [comment]
> Cool feature, I dont have time to do a good review until next week. Maybe @helto4real can check it out?

Really, I had to build it for my need of structure. It's just my OCD when it comes to project files.

File: src/HassModel/NetDaemon.HassModel.CodeGenerator/Extensions/CodeGeneratorExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace NetDaemon.HassModel.CodeGenerator.Extensions
+{
+    internal static class CodeGeneratorExtensions
+    {
+        public static string GetClassNameFromCompilationUnit(this CompilationUnitSyntax compilationUnit)
```
Comment (helto4real) [inline src/HassModel/NetDaemon.HassModel.CodeGenerator/Extensions/CodeGeneratorExtensions.cs:11]: GetClassName should be sufficient. We allready know it is a CompilationUnit

File: src/HassModel/NetDaemon.HassModel.CodeGenerator/CodeGenerationSettings.cs [hunk 2]
```diff
@@ -3,5 +3,7 @@
 public class CodeGenerationSettings
 {
     public string OutputFile { get; init; } = "HomeAssistantGenerated.cs";
+    public string OutputFolder { get; init; } = string.Empty;
     public string Namespace { get; init; } = "HomeAssistantGenerated";
+    public bool FilePerEntity { get; set; }
```
Comment (helto4real) [inline src/HassModel/NetDaemon.HassModel.CodeGenerator/CodeGenerationSettings.cs:8]: Think booleans should be verb, suggestion "GenerateOneFilePerEntity".

**Review by helto4real — COMMENTED** [review]
Thank you for your contribution. This was a gread idea I think! I think we are good to go with some small suggestions.

**Comment by helto4real** [comment]
@jtolar do you still want to merge this PR? There are some suggestions that I would like you to look at please.

**Comment by helto4real** [comment]
> @jtolar do you still want to merge this PR? There are some suggestions that I would like you to look at please.

Not it is my time to say "I am sorry" I totally missed you made the changes. :) I will review it again!

**Comment by helto4real** [comment]
@jtolar you want to finish this PR or should I close it? Thanks!

**Comment by jtolar** [comment]
I am not getting emails when comments are added for some reason. Let me see about resolving the conflicts.

**Review by helto4real — APPROVED** [review]
Thanks alot!

**Pull request merged**